addons/website_sale_stock/models/product_product.py

```diff
@@ -28,7 +28,7 @@ def _get_cart_qty(self, website=None):
 
     def _is_sold_out(self):
         combination_info = self.with_context(website_sale_stock_get_quantity=True).product_tmpl_id._get_combination_info(product_id=self.id)
-        return combination_info['product_type'] == 'product' and combination_info['free_qty'] <= 0
+        return combination_info['product_type'] == 'product' and combination_info['free_qty'] <= 0 and not combination_info.get('prevent_zero_price_sale')
```

Comment:
The function is about a product being sold out, not about the product price being 0..

Reply:
We should not show out of stock message for products with price=0. should I just hide the message in the xml template instead of changing this function?